tests/unit/samplers/test_ns_aw_validation.py

```
@@ -0,0 +1,83 @@
+    lp = _flat_uniform_prior(2)
+    sampler = _make_sampler(lp)
+    assert sampler.n_dims == 2
+
+
+def test_non_flat_prior_raises():
+    """A Gaussian prior (not flat) should raise ValueError."""
+
+    def gaussian_prior(arr):
+        return -0.5 * jnp.sum(arr**2)
+
+    with pytest.raises(ValueError, match="must return 0.0 for all points"):
```

Ruff (0.15.12)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)